Core/include/Acts/Propagator/Propagator.hpp

```diff
@@ -46,11 +46,11 @@ struct PropagatorResult : private detail::Extendable<result_list...> {
   /// Accessor to additional propagation quantities
   using detail::Extendable<result_list...>::get;
 
-  /// Final track parameters - initialized to null pointer
-  std::unique_ptr<parameters_t> endParameters = nullptr;
+  /// Final track parameters
+  std::optional<parameters_t> endParameters = std::nullopt;
```

Comment:
I think you could drop the `std::nullopt` since this is the default constructed value. but it may be more consistent across the repo to have it

Reply:
Yeah I left it because the same argument in principle works for the `std::unique_ptr`, and I think it is there for explcitness (if thats a word^^)... But you're right, maybe I drop it